internal/events/subscription.go

```diff
@@ -237,6 +253,29 @@ func (s *subscription) getEventTimestamp(ctx context.Context, l *logEntry) {
 	s.lp.stream.blockTimestampCache.Add(blockNumber, l.Timestamp)
 }
 
+func (s *subscription) getTransactionInputs(ctx context.Context, l *logEntry) {
+	abi, err := loadABI(s.cr, s.info.ABI)
```

Comment:
This is obviously pretty heavyweight - for every event, we fetch the ABI and then iterate over all the methods to determine which one was called. Would appreciate thoughts on the best ways and places to cache things. Perhaps a method cache on `ContractResolver`?

Reply:
Yeah, I think a method ID based cache on ContractResolver is important. I agree this could be very heavyweight without that.

There's a nice and flexible LRU cache facility used in FireFly core here:
https://github.com/hyperledger-labs/firefly/blob/c9d070f5f5b9ee07842fb1cc1cb8baec54231f9e/internal/privatemessaging/groupmanager.go#L152-L156

Reply:
That looks to be using https://github.com/karlseguin/ccache
Here in ethconnect, eventstream is already using https://github.com/hashicorp/golang-lru

Seems like we'd only want one per project, so I'll probably start with the hashicorp one unless there's a good reason to switch.

Reply:
Sounds good. My recollection for the reason we used `ccache` in FireFly, was that `golang-lru` didn't support sizing the objects (in Kb) - only a count.

Here only a count seems fine, so agree with sticking for consistency.